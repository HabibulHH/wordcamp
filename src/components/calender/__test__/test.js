import { screen } from "@testing-library/dom";
import { cleanup, render } from "@testing-library/react";
import axios from "axios";
import React from "react";
import { unmountComponentAtNode } from "react-dom";
import Calender from "../";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

jest.mock("axios");

afterEach(cleanup);

it("renders calender correctly", async () => {
  const fakeEvents = [
    {
      title: "Mock title",
      start: "1624492800",
      end: "1624665600",
      id: "34343444",
      location: "USA",
      url: "https:demo/link#3w4/s",
    },
  ];
  axios.get.mockResolvedValue({
    data: fakeEvents,
    headers: { "x-wp-totalpages": 11 },
  });
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, "useState");
  useStateSpy.mockImplementation((init) => [init, setState]);
  const { getByTestId, asFragment } = render(<Calender />);
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId("spinner"));
});
