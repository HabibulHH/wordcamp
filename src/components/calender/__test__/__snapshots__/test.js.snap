// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calender correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="spinner"
    >
      <div
        class="spinner-grow text-secondary"
      />
      <div
        class="spinner-grow text-success"
      />
      <div
        class="spinner-grow text-danger"
      />
      <div
        class="spinner-grow text-warning"
      />
      <div
        class="spinner-grow text-info"
      />
      <div
        class="spinner-grow text-light"
      />
      <div
        class="spinner-grow text-dark"
      />
    </div>
    <div>
      <div
        class="rbc-calendar"
        style="height: 600px;"
      >
        <div
          class="rbc-toolbar"
        >
          <span
            class="rbc-btn-group"
          >
            <button
              type="button"
            >
              Today
            </button>
            <button
              type="button"
            >
              Back
            </button>
            <button
              type="button"
            >
              Next
            </button>
          </span>
          <span
            class="rbc-toolbar-label"
          >
            September 2021
          </span>
          <span
            class="rbc-btn-group"
          >
            <button
              class="rbc-active"
              type="button"
            >
              Month
            </button>
            <button
              class=""
              type="button"
            >
              Week
            </button>
            <button
              class=""
              type="button"
            >
              Work Week
            </button>
            <button
              class=""
              type="button"
            >
              Day
            </button>
            <button
              class=""
              type="button"
            >
              Agenda
            </button>
          </span>
        </div>
        <div
          aria-label="Month View"
          class="rbc-month-view"
          role="table"
        >
          <div
            class="rbc-row rbc-month-header"
            role="row"
          >
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sun
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Mon
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Tue
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Wed
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Thu
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Fri
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sat
              </span>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content rbc-row-content-scrollable"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    29
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    30
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    31
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    01
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    02
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    03
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    04
                  </a>
                </div>
              </div>
              <div
                class="rbc-row-content-scroll-container"
              />
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content rbc-row-content-scrollable"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    05
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    06
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    07
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    08
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    09
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    10
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    11
                  </a>
                </div>
              </div>
              <div
                class="rbc-row-content-scroll-container"
              />
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content rbc-row-content-scrollable"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    12
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    13
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    14
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    15
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    16
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    17
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    18
                  </a>
                </div>
              </div>
              <div
                class="rbc-row-content-scroll-container"
              />
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content rbc-row-content-scrollable"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    19
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    20
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    21
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    22
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    23
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    24
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    25
                  </a>
                </div>
              </div>
              <div
                class="rbc-row-content-scroll-container"
              />
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg rbc-today"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
            </div>
            <div
              class="rbc-row-content rbc-row-content-scrollable"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    26
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-now rbc-current"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    27
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    28
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    29
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    30
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    01
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    02
                  </a>
                </div>
              </div>
              <div
                class="rbc-row-content-scroll-container"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
